src/main/java/appeng/core/sync/packets/PacketMultiPart.java

```diff
@@ -20,45 +20,48 @@
 
 
 import io.netty.buffer.ByteBuf;
-import io.netty.buffer.Unpooled;
 
-import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraftforge.common.MinecraftForge;
 
+import cpw.mods.fml.common.network.simpleimpl.MessageContext;
+
 import appeng.core.sync.AppEngPacket;
-import appeng.core.sync.network.INetworkInfo;
+import appeng.core.sync.AppEngPacketHandler;
+import appeng.helpers.Reflected;
 import appeng.integration.IntegrationRegistry;
 import appeng.integration.IntegrationType;
 import appeng.integration.abstraction.IFMP;
 
 
-public class PacketMultiPart extends AppEngPacket
+public class PacketMultiPart implements AppEngPacket, AppEngPacketHandler<PacketMultiPart, AppEngPacket>
 {
 
-	// automatic.
-	public PacketMultiPart( final ByteBuf stream )
-	{
-	}
-
-	// api
+	@Reflected
 	public PacketMultiPart()
 	{
-		final ByteBuf data = Unpooled.buffer();
-
-		data.writeInt( this.getPacketID() );
-
-		this.configureWrite( data );
+		// automatic.
 	}
 
 	@Override
-	public void serverPacketData( final INetworkInfo manager, final AppEngPacket packet, final EntityPlayer player )
+	public AppEngPacket onMessage( final PacketMultiPart message, final MessageContext ctx )
 	{
 		final IFMP fmp = (IFMP) IntegrationRegistry.INSTANCE.getInstance( IntegrationType.FMP );
 		if( fmp != null )
 		{
-			final EntityPlayerMP sender = (EntityPlayerMP) player;
+			final EntityPlayerMP sender = (EntityPlayerMP) ctx.getServerHandler().playerEntity;
 			MinecraftForge.EVENT_BUS.post( fmp.newFMPPacketEvent( sender ) ); // when received it just posts this event.
 		}
+		return null;
+	}
+
+	@Override
+	public void fromBytes( final ByteBuf buf )
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1186)